File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        max_profit = 0
        min_buy = prices[0]
        for i in range(len(prices)):
            min_buy = min(min_buy, prices[i])
            max_profit = max(max_profit, prices[i] - min_buy)
                    
        return max_profit

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.go
func maxProfit(prices []int) int {
    maxProfit := 0
    minBuy := prices[0]
    for _, v := range prices {
        if v < minBuy {
            minBuy = v
        }
        if v - minBuy > maxProfit {
            maxProfit = v - minBuy
        }
    }
    return maxProfit
}